onnxruntime/core/providers/qnn/builder/qnn_utils.cc

```diff
@@ -63,6 +65,12 @@ size_t GetElementSizeByType(ONNXTensorElementDataType elem_type) {
   return pos->second;
 }
 
+size_t GetQnnTensorDataSize(gsl::span<const uint32_t> shape, Qnn_DataType_t element_type) {
+  ORT_ENFORCE(!shape.empty(), "Empty shape not allowed.");  // TODO can we just treat empty shape as a scalar?
```

Comment:
this check is copied from the original implementation here:
https://github.com/microsoft/onnxruntime/blob/31e6e1010c9a51ba908f01fd03cf01cd55a75b83/onnxruntime/core/providers/qnn/builder/qnn_model.cc#L281

I'm not sure if it's needed